api/server/router/container/container.go

```diff
@@ -71,6 +71,6 @@ func (r *containerRouter) initRoutes() {
 		// PUT
 		router.NewPutRoute("/containers/{name:.*}/archive", r.putContainersArchive),
 		// DELETE
-		router.NewDeleteRoute("/containers/{name:.*}", r.deleteContainers),
+		router.NewDeleteRoute("/containers/{name:[^/]*}", r.deleteContainers),
```

Comment:
I think in order to leave this the same, I think you have to register the experimental routes for the checkpoint before the other container routes.  This is because the more specific routes should be added first before more generic ones.  I think you had to change this because its more generic than the checkpoint route.  You can keep this the same as long as you register the checkpoint routes first and then we won't have a change in functionality.